### ISSUE TITLE
Multiple entries in herd_tracking gives duplicates in individuals api call

### ISSUE THREAD

**Issue by pontus** [NBISweden/theherdbook#328]
R-api for hopefully current data shows in logs 

```
Duplicated IDs were removed.
This includes e.g.
         Indiv      Sire       Dam    Sex Born is_active
8636 G684-0911 G529-0732 G529-0741 female 2009     FALSE
9280 G684-0912 G529-0732 G529-0741 female 2009     FALSE
9282 G684-0913 G529-0732 G529-0741 female 2009     FALSE
```

I believe this may be related to

```
herdbook=# select * from herd_tracking where individual_id=8636;
 herd_tracking_id | from_herd_id | herd_id | signature_id | individual_id | herd_tracking_date 
------------------+--------------+---------+--------------+---------------+--------------------
             8636 |              |     406 |              |          8636 | 2008-06-30
            15860 |              |     406 |              |          8636 | 2008-12-31
            16483 |              |     406 |              |          8636 | 2009-12-31
            17136 |              |     406 |              |          8636 | 2010-12-31
(4 rows)

herdbook=# 
```

but am not certain; for the other individuals we have

```
herdbook=# select * from herd_tracking where individual_id=9280;
 herd_tracking_id | from_herd_id | herd_id | signature_id | individual_id | herd_tracking_date 
------------------+--------------+---------+--------------+---------------+--------------------
             9280 |              |     422 |              |          9280 | 2019-04-29
            27042 |              |     422 |              |          9280 | 2020-12-31
(2 rows)

herdbook=# select * from herd_tracking where individual_id=9282;
 herd_tracking_id | from_herd_id | herd_id | signature_id | individual_id | herd_tracking_date 
------------------+--------------+---------+--------------+---------------+--------------------
             9282 |              |     422 |              |          9282 | 2019-05-03
            25949 |              |      36 |              |          9282 | 2019-12-31
            27043 |              |      36 |              |          9282 | 2020-12-31
(3 rows)

herdbook=# 


```

Can we see what would cause the query at https://github.com/NBISweden/theherdbook/blob/develop/app/utils/data_access.py#L1074 to give the same individual multiple times?

**Comment by pontus** [comment]
Upon further inspection I think the herd tracking may be a red herring, e.g. 8636 only shows up once as an individual in the API - but as male (and as father). So why does optisel conclude 8636 is female?

**Comment by andkaha** [comment]
> Upon further inspection I think the herd tracking may be a red herring, e.g. 8636 only shows up once as an individual in the API - but as male (and as father). So why does optisel conclude 8636 is female?

"8636" is a `herd_tracking_id`, not an `individual_id`. The `individual_id` associated with `G684-0911` is "9468", which is a female.

And for that id:
```
herdbook=# select * from herd_tracking where individual_id = 9468;
 herd_tracking_id | from_herd_id | herd_id | signature_id | individual_id | herd_tracking_date
------------------+--------------+---------+--------------+---------------+--------------------
             9468 |              |     435 |              |          9468 | 2009-12-31
            16593 |              |     435 |              |          9468 | 2009-12-31
(2 rows)
```

**Comment by andkaha** [comment]
The issue seems to be the query at https://github.com/NBISweden/theherdbook/blob/a00157f22b22dbc4f1d14442fd53c64d2d5cf274/app/utils/data_access.py#L1038

For the problematic individuals, this returns
```
 herd_id | herd_tracking_date | individual_id | rank
---------+--------------------+---------------+------
[...]
     435 | 2009-12-31         |          9468 |    1
     435 | 2009-12-31         |          9468 |    1
     435 | 2009-12-31         |          9469 |    1
     435 | 2009-12-31         |          9469 |    1
     435 | 2009-12-31         |          9470 |    1
     435 | 2009-12-31         |          9470 |    1
```

I would expect the "rank" column to be alternating 1, 2 , 1, 2, 1, 2.

I'm currently trying to understand this.

**Comment by andkaha** [comment]
I think it comes back to individuals having more than a single event in the `herd_tracking` table.  To me, having multiple entries in the `herd_tracking` table doesn't sound like something that should break the API. A rabbit should be able to be sold between herds more than once per year, this does not sound far fetched. I'm therefore looking at possible solutions for correcting the API (a modified query, possibly) instead of at how to fudge the data so that the broken API works.

**Comment by andkaha** [comment]
I wonder if this is solved by simply making the above query a `SELECT DISTINCT` query?

**Comment by jhagberg** [comment]
@kusalananda should this issue be closed as its fixed by #328?

**Comment by andkaha** [comment]
> @kusalananda should this issue be closed as its fixed by #328?

By #330 , yes. Closing.
